### PR TITLE
add desensitization extension

### PR DESCRIPTION
**Pull request by xzchaoo**
### Motivation:

data security

### Modification:

add desensitization extension

### CODE REVIEW

File: tracer-core/src/main/java/com/alipay/common/tracer/util/CoolDown.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+package com.alipay.common.tracer.util;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * 封装一个定时重置的计数器用于降低某些非必要操作的频率(比如某些预期内的错误日志).
+ * 具有如下特征:
+ * <ul>
+ *     <li>计数器的值永远不超过times</li>
```
Comment (guanchao-yang) [inline tracer-core/src/main/java/com/alipay/common/tracer/util/CoolDown.java]: The comment cloud be English

**Review by guanchao-yang — CHANGES_REQUESTED** [review]

File: tracer-core/src/test/java/com/alipay/common/tracer/util/CoolDownTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public void test() throws Throwable {
+        int times = 5;
+        int periods = 3;
+        final CoolDown coolDown = new CoolDown(1000, times);
+        ExecutorService es = Executors.newFixedThreadPool(4);
+        List<Future<?>> futures = new ArrayList<Future<?>>();
+
+        final AtomicInteger counter = new AtomicInteger();
+        long start = System.currentTimeMillis();
+        // 在 1000ms 和 2000ms 时刻 会重置
```
Comment (guanchao-yang) [inline tracer-core/src/test/java/com/alipay/common/tracer/util/CoolDownTest.java]: English comment

File: tracer-core/src/main/java/com/alipay/common/tracer/util/CoolDown.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        this.intervalMills = resetIntervalMills;
+        this.times = times;
+        this.nextResetTime = System.currentTimeMillis() + resetIntervalMills;
+    }
+
+    boolean tryAcquire() {
+        while (true) {
+            int c = counter.get();
+            if (c < times) {
+                if (counter.compareAndSet(c, c + 1)) {
```
Comment (guanchao-yang) [inline tracer-core/src/main/java/com/alipay/common/tracer/util/CoolDown.java:45]: When can‘t get the  CAS lock  would be retry ,why not return false?

Comment (xzchaoo) [reply tracer-core/src/main/java/com/alipay/common/tracer/util/CoolDown.java:45]: Like
1. java.util.concurrent.atomic.AtomicInteger#getAndIncrement
2. sun.misc.Unsafe#getAndAddInt

Every time, at least one thread will success.

File: tracer-core/src/test/java/com/alipay/common/tracer/util/CoolDownTest.java (hunk 2 above)
Comment (xzchaoo) [reply tracer-core/src/test/java/com/alipay/common/tracer/util/CoolDownTest.java]: done

**Review by guanchao-yang — APPROVED** [review]

**Review by glmapper — APPROVED** [review]
LGTM

**Pull request merged**